app/src/main/java/com/sdp/swiftwallet/CryptoValuesActivity.java

```diff
@@ -90,43 +95,73 @@ private void addToCurrencyList(){
 
     private void getCurrencyData() {
         progressBar.setVisibility(View.VISIBLE);
-        String url = "https://pro-api.coinmarketcap.com/v1/cryptocurrency/listings/latest";
+        String url = "https://api.binance.com/api/v3/ticker/24hr";
         RequestQueue requestQueue = Volley.newRequestQueue(this);
-        JsonObjectRequest jsonObjectRequest = new JsonObjectRequest(Request.Method.GET, url, null, response -> {
+
+        JsonArrayRequest jsonArrayRequest = new JsonArrayRequest(Request.Method.GET, url, null, response -> {
             progressBar.setVisibility(View.GONE);
-            try {
-                JSONArray dataArray = response.getJSONArray("data");
-                for (int i = 0; i < dataArray.length(); i++) {
-                    JSONObject dataObject = dataArray.getJSONObject(i);
-                    String name = dataObject.getString("name");
+            try{
+                for(int i = 0; i<response.length(); ++i){
+                    JSONObject dataObject = response.getJSONObject(i);
                     String symbol = dataObject.getString("symbol");
-                    JSONObject quote = dataObject.getJSONObject("quote");
-                    JSONObject valueJSON = quote.getJSONObject("USD");
-                    double value = valueJSON.getDouble("price");
-                    currencyArrayList.add(new Currency(name, symbol, value));
+                    if(showUSDTOnly == "Show All"){
+                        String name = dataObject.getString("symbol");
+                        double value = dataObject.getDouble("lastPrice");
+                        currencyArrayList.add(new Currency(name, symbol, value));
+                    } else if(symbol.endsWith("USDT")) {
+                        String name = dataObject.getString("symbol");
+                        name = name.substring(0, name.length() - 4);
+                        double value = dataObject.getDouble("lastPrice");
+                        currencyArrayList.add(new Currency(name, symbol, value));
+                    }
                 }
                 currencyAdapter.notifyDataSetChanged();
                 mIdlingResource.decrement();
-            } catch (JSONException e) {
+
+            } catch(Exception e){
                 e.printStackTrace();
                 Toast.makeText(CryptoValuesActivity.this, "Couldn't extract JSON data... Please try again later.", Toast.LENGTH_SHORT).show();
             }
-
         }, error -> {
-            progressBar.setVisibility(View.GONE);
             Toast.makeText(CryptoValuesActivity.this, "Couldn't retrieve data... Please try again later.", Toast.LENGTH_SHORT).show();
+        });
+
+
+        requestQueue.add(jsonArrayRequest);
+    }
+
+    /*  CAN BE USED LATER TO SET A SPINNER THAT ALLOWS USER TO DECIDE IF THEY WANT TO
+        ADDED TO EYMERIC'S SPRINT 9 TASK
+
```

Comment:
I think it's better to remove it and put it in another developpment branch if it's not used for now